custom_components/frigate/update.py

```diff
@@ -0,0 +1,102 @@
+"""Update platform for frigate."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.update import UpdateEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_MODEL, CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import FrigateDataUpdateCoordinator, get_frigate_device_identifier
+from .const import ATTR_COORDINATOR, DOMAIN, NAME
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Sensor entry setup."""
+    coordinator = hass.data[DOMAIN][entry.entry_id][ATTR_COORDINATOR]
+
+    entities = []
+    entities.append(FrigateContainerUpdate(coordinator, entry))
+    async_add_entities(entities)
+
+
+class FrigateContainerUpdate(UpdateEntity, CoordinatorEntity):  # type: ignore[misc]
+    """Frigate container update."""
+
+    _attr_title = "Frigate Container"
+
+    def __init__(
+        self,
+        coordinator: FrigateDataUpdateCoordinator,
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Construct a FrigateContainerUpdate."""
+        CoordinatorEntity.__init__(self, coordinator)
+        self._config_entry = config_entry
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID to use for this entity."""
+        return "frigate_container"
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Get device information."""
+        return {
+            "identifiers": {get_frigate_device_identifier(self._config_entry)},
+            "name": NAME,
+            "model": self._get_model(),
+            "configuration_url": self._config_entry.data.get(CONF_URL),
+            "manufacturer": NAME,
+        }
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return "Frigate Container"
+
+    @property
+    def installed_version(self) -> str | None:
+        """Version currently in use."""
+
+        version_hash = self.coordinator.data.get("service", {}).get("version")
+
+        if not version_hash:
+            _LOGGER.error("Version not contained in stats")
```

Comment:
How often is this method called? Could this spam the logs?

Reply:
https://developers.home-assistant.io/docs/core/entity/update
I can't find it now but I thought I read somewhere it isn't too frequently. If we are worried about log spam I could remove it since it isn't necessary relevant to a user since it doesn't necessarily break anything?

Reply:
Yeah, I think just remove it. The sensor just wouldn't do anything.